include/libtorrent/aux_/portmap.hpp

```diff
@@ -66,6 +66,32 @@ namespace libtorrent { namespace aux {
 	protected:
 		~portmap_callback() {}
 	};
+
+		struct base_mapping
+	{
+		enum class action : std::uint8_t { none, add, del };
```

Comment:
please hoist this enum class out of the ``base_mapping`` struct. I think it should belong with the other enum classes (``portmap_protocol`` and ``portmap_transport``). Would it cause a lot of trouble to do this? perhaps ``action`` is a bit too broad of a name. ``portmap_action`` seems to fit in with the other enums. what do you think?

my thinking is that since it's a proper type (enum class) it doesn't need to be "potected" by being scoped by the class